### PR TITLE
ASB-689: Update OpenAPI spec with Submissions API

### PR DESCRIPTION
**Pull request by JamesP-KC**
- Document new POST submissions API endpoint in the OpenAPI spec

### CODE REVIEW

**Review by mauriciokin — APPROVED** [review]
legit

File: packages/applications-service-api/api/openapi.yaml [hunk 1]
```diff
@@ -1006,4 +1039,97 @@ components:
           example: 1
         currentPage:
           type: number
-          example: 1
\ No newline at end of file
+          example: 1
+    SubmissionCreateRequestBody:
+      properties:
+        file:
```
Comment (Philip-Philippou1) [inline packages/applications-service-api/api/openapi.yaml:1045]: Do we want an MD5/size to be sent in the request as a way of validating the file at both ends of transfer.  Perhaps we don't.

File: packages/applications-service-api/api/openapi.yaml [hunk 2]
```diff
@@ -1006,4 +1039,97 @@ components:
...
+          description: Interested party reference number
+          example: 3287498209
+        deadline:
+          type: string
+          description: Name/title of Deadline
+          example: Deadline 1A
+        submissionType:
+          type: string
+          example: "Other: Biodiversity Ecology and natural Environment"
+        representation:
```
Comment (Philip-Philippou1) [inline packages/applications-service-api/api/openapi.yaml:1110]: Is the idea that if it's a comment we would send it using this endpoint as text and then the API can convert to PDF? 
That would work but we may need to remove the 'required' indicator from file.

Comment (JamesP-KC) [reply packages/applications-service-api/api/openapi.yaml]: Yes good call, I have pushed an update making `representation` and `file` optional but have added a description to try and highlight that one of them is required. There doesn't seem to be a non-hacky way of defining such requirements in OpenAPI

File: packages/applications-service-api/api/openapi.yaml [hunk 3]
```diff
@@ -1006,4 +1039,97 @@ components:
...
+          type: string
+          description: Interested party reference number
+          example: 3287498209
+        deadline:
+          type: string
+          description: Name/title of Deadline
+          example: Deadline 1A
+        submissionType:
+          type: string
+          example: "Other: Biodiversity Ecology and natural Environment"
```
[the remark this answers is not in the document]
Comment (JamesP-KC) [reply packages/applications-service-api/api/openapi.yaml]: Strange, it was rendering OK for me in editor.swagger.io. In any case, I've changed it to another example without quotes :)

**Review by Philip-Philippou1 — APPROVED** [review]

**Pull request merged**